packages/schematics/angular/library/index.ts

```diff
@@ -171,7 +171,11 @@ export default function (options: LibraryOptions): Rule {
     if (!options.name) {
       throw new SchematicsException(`Invalid options, "name" is required.`);
     }
-    const name = options.name;
+    // If scoped project (i.e. "@foo/bar"), convert projectDir to "foo-bar".
+    const packageName = options.name;
+    options.name = /@[\S]*\/[\S]*/.test(options.name)
```

Comment:
What if the name passed on the command line is `a/b/c`? Is it going to create `c` in `/projects/a/b`?